### PR TITLE
Simplify workflows and dockerfiles

### PR DESCRIPTION
**Pull request by sanekto**
With this change there is only one workflow that builds docker images. Also only one Dockerfile.

@dharmendrakariya The kubernetes config must have a new environment variable defined for the production deployment:
`PBA_ENV=production`

That way the run.sh script will run `yarn start` instead of `yarn dev`

### CODE REVIEW

**Comment by sanekto** [comment]
Is there anything blocking this?

**Comment by dharmendrakariya** [comment]
Hey @sanekto  overall it looks good to me, specially the dockerfile changes but for CI we want one more tag , branch name as tag! please add `type=ref,event=branch` and we are good! 

Also would like to ask @sgzerbo  if he is okay with this logic (reason I have duplication of Dockerfiles in all repos is that we want to keep everything way simpler and easy to understand for each tech guy) I believe this is also simple but just want his confirmation and we can merge it!

File: Dockerfile
Comment (dharmendrakariya) [inline Dockerfile:1]: We need to copy the run.sh here

File: .github/workflows/docker.yml
Comment (dharmendrakariya) [inline .github/workflows/docker.yml:1]: We have to add tag (branch name)

**Review by dharmendrakariya — CHANGES_REQUESTED** [review]
after the required changes, we are good!

**Review by dharmendrakariya — APPROVED** [review]

**Review by ismaelga — APPROVED** [review]

**Comment by dharmendrakariya** [comment]
@sanekto  @sgzerbo  we can merge this! I have tested this images and working fine!

**Comment by sanekto** [comment]
Great, thanks for taking care. 

The last commits show why I beleive using github actions (docker/metadata-action@v5, actions/checkout@v4, etc) is flawed, even if github promotes it. It is something that really adds confusion and it is not locally reproducible so it adds a lot of work and debugging time instead of reducing it. And it also does vendor lock-in (that is why I believe github promotes it). 
For the future I suggest to do a script based approach, something like https://github.com/karpatkey/roles_royce/blob/main/.github/scripts/build_and_push.sh . It is easier to reproduce locally (just provide a couple of env variables) and there is no "magic" just a plain script that everyone can read and understand. Even if we add all the cool things like caching etc it is in something approachable.

**Comment by dharmendrakariya** [comment]
@sanekto  bro totally agree with you, just to add the simple branch name as a tag I was struggling and there is no way we can debug this locally! I like your idea! We shall move to the shell scripting! I saw your changes in roles_royce! Similar approach we can adapt here! wdyt?

**Comment by sanekto** [comment]
Yes I think a similar approach can be done here (in another PR). 
Merge this one whenever you like.

**Comment by ismaelga** [comment]
@sanekto @dharmendrakariya I actually think we could run prod build in these preview envs, meaning that way we wouldn't need that script and the new env var

**Comment by sanekto** [comment]
I don't know the diff between the two versions, I never run the app myself. So I am +1 on whatever you think is best.

**Comment by dharmendrakariya** [comment]
@ismaelga  we want to run develop images in PR-review deployments! could you create a separate PR with your commits bro?
I think we shall merge this branch without your changes as of now!

**Comment by ismaelga** [comment]
@dharmendrakariya I'll remove my changes but keep in mind the build wasn't working properly. You still need to fix it after

**Pull request merged**